tp_framework/core/discovery.py

```diff
@@ -257,105 +246,367 @@ def discovery(src_dir: Path, l_tp_id: list[int], tp_lib_path: Path, itools: list
+            measurement_stop: bool = d_tpi["measurement"] not in ["ignore", "not_supported"]
+            d_tpi["discovery"] = discovery_for_tpi(tpi_instance, tpi_json_path, cpg, disc_output_dir,
+                                                   measurement_stop=measurement_stop, already_executed=d_dr_executed)
+            d_res_tpi[tpi_id] = d_tpi
+        d_res[tp_id]["instances"] = d_res_tpi
+
+    # post-process results and export them
+    d_results = post_process_and_export_results(d_res, build_name, disc_output_dir)
+    d_results["tools"] = tools
+    d_results["cpg_file"] = str(cpg)
+    logger.info("Discovery for patterns completed.")
+    return d_results
 
-    for f in findings_for_tools:
-        f["instanceId"] = (''.join(f'{iid}, ' for iid in f["instanceId"]))[:-2]
 
-    findings: list[Dict] = [dict(t) for t in {tuple(d.items()) for d in findings_for_tools}]
+def discovery_ignore_measurement(cpg: Path, l_tp_id: list[int], tp_lib: Path,
+                                 language: str,
+                                 build_name: str,
+                                 disc_output_dir: Path,
+                                 timeout_sec: int = 0) -> Dict:
+    d_res = {}  # result data structure set to empty
+    # loop over testability patterns (tp) to be discovered
+    for tp_id in l_tp_id:
+        d_res[tp_id] = {"measurement_found": None}
+        l_tpi_jsonpath = utils.list_tpi_paths_by_tp_id(language, tp_id, tp_lib)  # TODO: do we need this later?
+        d_tpi_id_path = {}
+        for jp in l_tpi_jsonpath:
+            d_tpi_id_path[utils.get_tpi_id_from_jsonpath(jp)] = jp
+        # loop over tp instances (tpi) to be discovered
+        d_res_tpi = {}
+        d_dr_executed = {}
+        for tpi_id in d_tpi_id_path:
+            tpi_json_path = d_tpi_id_path[tpi_id]
+            tpi_json_rel = os.path.relpath(tpi_json_path, start=tp_lib)
+            tpi_instance = load_instance_from_metadata(tpi_json_rel, tp_lib, language)  # get the instance
+            d_tpi = {"instance": tpi_instance, "measurement": "ignored", "jsonpath": tpi_json_path,
+                     "discovery": discovery_for_tpi(tpi_instance, tpi_json_path, cpg, disc_output_dir,
+                                                    measurement_stop=False, already_executed=d_dr_executed)}
+            d_res_tpi[tpi_id] = d_tpi
+        d_res[tp_id]["instances"] = d_res_tpi
+
+    # post-process results and export them
+    d_results = post_process_and_export_results(d_res, build_name, disc_output_dir)
+    d_results["cpg_file"] = str(cpg)
+    logger.info("Discovery for patterns completed.")
+    return d_results
+
 
+def discovery_for_tpi(tpi_instance: Instance, tpi_json_path: Path, cpg: Path, disc_output_dir: Path,
+                      measurement_stop: bool = False, already_executed: dict = {}) -> Dict:
+    msgpre = f"pattern {tpi_instance.pattern_id} instance {tpi_instance.instance_id} - "
+    d_tpi_discovery = {}

```

Comment:
```suggestion
    d_tpi_discovery = dict.fromkeys(["rule_path", "method", "rule_name", "rule_accuracy", "rule_hash", \
                                     "rule_name", "results", "rule_already_executed"], None)
```
This makes the code a bit shorter and assures, that the dict contains all necessary keys in all branching cases